cli/cmdlineparser.cpp

```diff
@@ -78,7 +78,7 @@ static bool addFilesToList(const std::string& fileList, std::vector<std::string>
     return true;
 }
 
-static bool addIncludePathsToList(const std::string& fileList, std::list<std::string>* pathNames)
+static bool addIncludePathsToList(const std::string& fileList, std::list<std::string>& pathNames)
```

Comment:
this is my programming style.

If you call `addIncludePathsToList(fileList, pathNames)` .. both parameters by value (you can not see it's passed by reference looking at the call).. then that means the `addIncludePathsToList` will not modify pathNames.

Since addIncludePathsToList modifies pathNames I think it's more explicit to write `addIncludePathsToList(fileList, &pathNames)`.

Reply:
This is not a coding style decision. This is about safety and proper C++. If the parameter is not optional you might pass a `nullptr` (which we do but fortunately only is like three cases) and it will be dereferenced.

If this would be our style than we would have to convert **every** modifying reference to a pointer and also add checking of the pointer to each and every call. That would be a mess and would probably also hurt performance in a big way.

Reply:
Thinking about it a bit more this style doesn't really makes sense since you will only see this on the very first call and not subsequent ones in the chain.